part-3/part-3.md

```diff
@@ -152,252 +153,293 @@ const source = from([1, 2, 3]).pipe(
 
-// Выведет в консоль:
+// Will log:
 
-//. [0, 0] через 0.5 sec
+//. [1, 0] in 1 second
 
-//. [1, 0] через 1.0 sec
+//. [2, 0] in 1.5 seconds
 
-// [1, 1] через 1.5 sec
+// [3, 1] in 2 seconds
 
-// [2, 1] через 2.0 sec
+// and so on
 
-source.subscribe((value) => console.log(value));
+source.subscribe(value => console.log(value));
 ```
 
-Смотрите! Не смотря на то, что значения в потоках генерируются в разное время, нам удалось их свести к одному значению, что позволяет нам одновременно обрабатывать сразу два значения из потока.
+Despite the values of these two streams are generated at different times, we manage to combine them into a single value which allows us to handle values from two different streams simultaneously. **combineLatest** allows us to combine as many `Observables` as we want, not just two.
 
-## **withLastestFrom()**
+## **withLatestFrom()**
 
-Как вы могли заметить, мы получали новое значение каждый раз, когда значение в одном из потоков менялось. А что если, нам надо получать новое значение только тогда, когда поменялся основной поток? Давайте это сделаем с помощью оператора **withLatestFrom()**.
+As you could see, we received a new value each time any of the two streams emitted. But what if we need to receive a value only when the source `Observable` emits? Let's achieve that using the **withLatestFrom()** operator.
 
-**withLatestFrom()** комбинирует текущий поток с переданным потоком, и создает новое значение только тогда, когда оно добавилось в основном потоке.
+**withLatestFrom()** combines the source `Observable` with another one, and emits a combined value only when the source one emits.
 
 ```ts
-import { timer } from "rxjs";
+import { interval } from 'rxjs';
 
-import { withLatestFrom } from "rxjs/operators";
+import { withLatestFrom } from 'rxjs/operators';
 
-// Первое значение вернет сразу, а остальные каждую секунду
+const sourceFirst = interval(1000);
 
-const sourceFirst = timer(0, 1000);
-
-// Первое значение через 0.5 секунды, а остальные каждую секунду
-
-const sourceSecond = timer(500, 1000);
+const sourceSecond = interval(500);
 
 const source = sourceFirst.pipe(withLatestFrom(sourceSecond));
 
-// Выведет в консоль:
+// Will log:
 
-//. [1, 0] через 1.0 sec
+//. [0, 1] in 1.0 second
 
-// [2, 1] через 2.0 sec
+// [1, 3] in 2.0 seconds
 
-// [3, 2] через 3.0 sec
+// [2, 5] in 3.0 seconds
 
-source.subscribe((value) => console.log(value));
+source.subscribe(value => console.log(value));
 ```
 
-Разница с предыдущим примером в том, что теперь новое значение создается только тогда, когда изменяется основной поток.
+The only difference from **combineLatest** is that it will only emit when the source `Observable` emits. But it won't emit if the `Observable` we proided to **withLatestFrom** has not emitted at all, even if the source `Observable` emits, just like **combineLatest**, so for instance, in this example, the first emission from `sourceSecond` will be ignored. 
```

Comment:
Does this sound correct, @Armenvardanyan95?
```suggestion
The only difference from **combineLatest** is that it will only emit when the source `Observable` emits. It won't emit if the `Observable` we passed to **withLatestFrom** has not emitted at all, even when the source `Observable` emits. This is similar to **combineLatest**, for instance in this example, the first emission from `sourceSecond` will be ignored. 
```